hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/contracts/ContractUtils.java

```diff
@@ -0,0 +1,176 @@
+            @NonNull final VirtualMapLike<VirtualBlobKey, VirtualBlobValue> fileStore,
+            @NonNull final Collection</*@NonNull*/ Integer> contractIds,
+            @NonNull final Collection</*@NonNull*/ Integer> deletedContractIds) {
+        Objects.requireNonNull(contractIds);
+        Objects.requireNonNull(deletedContractIds);
+
+        final var codes = new ArrayList<Contract>(ESTIMATED_NUMBER_OF_CONTRACTS);
+        for (final var cid : contractIds) {
+            final var vbk = new VirtualBlobKey(Type.CONTRACT_BYTECODE, cid);
+            if (fileStore.containsKey(vbk)) {
+                final var blob = fileStore.get(vbk);
+                if (null != blob) {
+                    final var c = new Contract(
+                            new TreeSet<>(),
+                            blob.getData(),
+                            deletedContractIds.contains(cid) ? Validity.DELETED : Validity.ACTIVE);
+                    c.ids().add(cid);
+                    codes.add(c);
+                }
+            }
+        }
+        return codes;
+    }
+
+    static Contracts getModContracts(HederaAccount[] dumpableAccounts) {
+        final var smartContracts = Arrays.stream(dumpableAccounts)
+                .filter(HederaAccount::smartContract)
+                .toList();
+        final var deletedSmartContract =
+                smartContracts.stream().filter(HederaAccount::deleted).toList();
+
+        final var extractedFiles = getContractStore();
+
+        final var contractContents = new ArrayList<Contract>(ESTIMATED_NUMBER_OF_CONTRACTS);
+        for (final var smartContract : smartContracts) {
+            final var contractId = getContractId(smartContract);
+            final var fileId = ContractID.newBuilder().contractNum(contractId).build();
+            if (extractedFiles.contains(fileId)) {
+                final var bytecode = extractedFiles.get(fileId);
+                if (null != bytecode) {
+                    final var c = new Contract(
+                            new TreeSet<>(),
+                            bytecode.code().toByteArray(),
+                            deletedSmartContract.contains(smartContract) ? Validity.DELETED : Validity.ACTIVE);
+                    c.ids().add(contractId);
+                    contractContents.add(c);
+                }
+            }
+        }
+
+        final var deletedContractIds =
+                deletedSmartContract.stream().map(ContractUtils::getContractId).toList();
+        return new Contracts(contractContents, deletedContractIds, smartContracts.size());
+    }
+
+    private static int getContractId(HederaAccount contract) {
+        if (contract.accountId() == null || contract.accountId().accountNum() == null) {
+            return 0;
+        }
+        return contract.accountId().accountNum().intValue();
+    }
+
+    private static ReadableKVState<ContractID, Bytecode> getContractStore() {
```

Comment:
⚠️ Unless I'm missing some magic here, isn't this just constructing an empty `KVState` and returning it?

Which means the mod service bytecodes will always appear to be missing?

I would have expected `dumpModContractBytecodes()` to receive e.g. a `VirtualMap<OnDiskKey<ContractID>, OnDiskValue<Bytecode>>` which was got [here](https://github.com/hashgraph/hedera-services/blob/dfc70e33db0bf73be47a51840e60bee686802061/hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/StateDumper.java#L115) via a call to 

```
state.getChild(
  state.findNodeIndex(ContractService.NAME,
  InitialModServiceContractSchema.BYTECODE_KEY)
```

Reply:
Is your last comment still valid?  I see some handling of deleted contracts by checking the accounts for the deleted flag.